react-components/src/components/RevealToolbar/SetFlexibleControlsType.tsx

```diff
@@ -13,86 +13,163 @@ import {
+
+type ControlTypeSelectionProps = {
+  selectedControlsType: FlexibleControlsType;
+  setSelectedControlsType: (controlsType: FlexibleControlsType) => void;
+  options: FlexibleControlsType[];
+  translateDelegate: TranslateDelegate;
+};
+
+export function SetOrbitOrFirstPersonControlsType(
+  props: Omit<CustomSettingsProps, 'includeOrbitInCenter'>
+): ReactElement {
+  return SetFlexibleControlsType({ ...props, includeOrbitInCenter: false });
 }
 
 export function SetFlexibleControlsType({
-  includeOrbitInCenter
+  includeOrbitInCenter,
+  topbarLayout
 }: CustomSettingsProps): ReactElement {
   const viewer = useReveal();
-  const manager = asFlexibleCameraManager(viewer.cameraManager);
+  const flexibleCameraManager = asFlexibleCameraManager(viewer.cameraManager);
   const { t: translate } = useTranslation();
 
   // This value is redunant (react force me to do it)
   const [selectedControlsType, setSelectedControlsType] = useState<FlexibleControlsType>(
-    getDefaultValue(manager)
+    getDefaultValue(flexibleCameraManager)
   );
-  const update = (newControlsType: FlexibleControlsType): void => {
-    setSelectedControlsType(newControlsType);
-  };
 
-  useEffect(() => {
-    if (manager === undefined) {
-      return;
-    }
-    manager.addControlsTypeChangeListener(update);
-    return () => {
-      if (manager !== undefined) {
-        manager.removeControlsTypeChangeListener(update);
+  const setSelectedControlsTypeAndUpdateCameraManager = useCallback(
+    (controlsType: FlexibleControlsType) => {
+      setSelectedControlsType(controlsType);
+      if (flexibleCameraManager !== undefined) {
+        flexibleCameraManager.controlsType = controlsType;
       }
-    };
-  }, []); // Should only be called once
+    },
+    [flexibleCameraManager]
+  );
 
-  if (manager === undefined) {
-    return <></>;
-  }
+  useListenToCameraManagerUpdate(flexibleCameraManager, setSelectedControlsType);
 
-  function getLabel(controlsType: FlexibleControlsType): string {
-    switch (controlsType) {
-      case FlexibleControlsType.FirstPerson:
-        return translate('CONTROLS_TYPE_FIRST_PERSON', 'Fly');
-      case FlexibleControlsType.Orbit:
-        return translate('CONTROLS_TYPE_ORBIT', 'Orbit');
-      case FlexibleControlsType.OrbitInCenter:
-        return translate('CONTROLS_TYPE_ORBIT_IN_CENTER', 'Center Orbit');
-      default:
-        return 'Undefined';
-    }
+  if (flexibleCameraManager === undefined) {
+    return <></>;
   }
 
   const options = [FlexibleControlsType.Orbit, FlexibleControlsType.FirstPerson];
+
   if (includeOrbitInCenter ?? false) {
     options.push(FlexibleControlsType.OrbitInCenter);
   }
 
+  const controlsTypeParameters: ControlTypeSelectionProps = {
+    selectedControlsType,
+    setSelectedControlsType: setSelectedControlsTypeAndUpdateCameraManager,
+    options,
+    translateDelegate: translate
+  };
+
+  return topbarLayout ?? false ? (
```

Comment:
Change this name. It is not dependent on using a Top tool bar. This  UI component can be used other places as well.